### PR TITLE
Async implementation of `WithLast`

### PR DESCRIPTION
**Pull request by FreeApophis**
* `AsyncSequence.Return` implemented

### CODE REVIEW

File: Funcky.Async/AsyncSequence/Return.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Funcky
+{
+    public static partial class AsyncSequence
+    {
+        public static IAsyncEnumerable<TItem> Return<TItem>(TItem item)
```
Comment (bash) [inline Funcky.Async/AsyncSequence/Return.cs:8]: `[Pure]` missing

File: Funcky.Async/Extensions/AsyncEnumerableExtensions/Intersperse.cs [hunk 2]
```diff
@@ -17,6 +13,7 @@ public static IAsyncEnumerable<TSource> Intersperse<TSource>(this IAsyncEnumerab
                 ? Return(item.Value)
                 : Return(element).Append(item.Value));
 
-        private static IAsyncEnumerable<TSource> Return<TSource>(TSource item) => AsyncEnumerable.Repeat(item, 1);
+        private static IAsyncEnumerable<TSource> Return<TSource>(TSource item)
+            => AsyncSequence.Return(item);
```
Comment (bash) [inline Funcky.Async/Extensions/AsyncEnumerableExtensions/Intersperse.cs:17]: I'd inline this method

**Review by bash — CHANGES_REQUESTED** [review]

File: Funcky.Async/Extensions/AsyncEnumerableExtensions/Intersperse.cs (hunk 2 above)
Comment (FreeApophis) [reply Funcky.Async/Extensions/AsyncEnumerableExtensions/Intersperse.cs]: Inlined in b0bb8e2ee87d98f7fbe1f837435da1d8f6824d77

File: Funcky.Async/AsyncSequence/Return.cs (hunk 1 above)
Comment (FreeApophis) [reply Funcky.Async/AsyncSequence/Return.cs:10]: Add Pure Attribute in fde87725a73eeeab576403f2b27ed2b373560e5a

**Review by bash — APPROVED** [review]

**Pull request merged**